core/src/main/scala/org/broadinstitute/dsde/rawls/entities/EntityService.scala

```diff
@@ -544,101 +550,204 @@ class EntityService(protected val ctx: RawlsRequestContext,
   }
 
   /**
-    * Migrate all entity data in a given workspace from legacy (LocalEntityProvider) to compact (Quicksilver) format.
-    *
-    * This migration method is unoptimized and in flux; use at your own risk
-    *
+    * Migrate all entity data in a given workspace from legacy (LocalEntityProvider) to
+    * compact (Quicksilver) format.
     */
-  def quicksilverMigration(workspaceName: WorkspaceName): Future[Map[String, Int]] = {
-    implicit val system: ActorSystem = ActorSystem("quicksilverMigration")
-
-    for {
-      // verify owner of workspace.
-      // TODO CORE-364: require some kind of admin permission via asFCAdmin or a resource type admin instead?
-      workspaceContext <- getV2WorkspaceContextAndPermissions(workspaceName,
-                                                              SamWorkspaceActions.own,
-                                                              Some(WorkspaceAttributeSpecs(all = false))
-      )
+  def quicksilverMigration(workspaceName: WorkspaceName): Future[Int] = {
+    // Number of entities to handle in a single chunk when migrating from legacy to compact.
+    // The migration relies on temp tables; this setting ensures the temp tables do not grow too large.
+    val batchSize = 50000
```

Comment:
I also tried 100k and 25k; both of those ended up being slower. There may be a more optimal number but I don't think it'll be a LOT faster.